api/src/multiaccessy/accessy.py

```diff
@@ -232,10 +240,41 @@ def get_member_id_from_accessy_id(self, accessy_id: UUID) -> Optional[int]:
     def get_all_members(self) -> list[AccessyMember]:
         """Get a list of all Accessy members in the ORG with GROUPS (lab and special)"""
 
-        org_member_ids = set(item["id"] for item in self._get_users_org())
+        def fill_in_group_affiliation_and_membership_id(members: list[AccessyMember], group: UUID) -> None:
+            group_members = self._get_users_in_access_group(group)
+            userId_to_group_member = {m.userId: m for m in group_members}
+            for member in members:
+                assert member.user_id is not None
+                membership = userId_to_group_member.get(member.user_id, None)
+
+                if not membership:
+                    continue
+
+                member.groups.add(group)
+                if member.membership_id is None:
+                    member.membership_id = membership.id
+                elif member.membership_id != membership.id:
+                    logger.error(
+                        f"User {member.name!r} has different membership_id for different groups. {member.user_id=}"
```

Comment:
I would raise an exception. This is critical enough. And we don't want to act on bogus data.

Reply:
Good point. Adding